### PR TITLE
EN-3257, add pass reason to page requested

### PR DESCRIPTION
**Pull request by nitzanpx**

### CODE REVIEW

File: src/main/java/com/perimeterx/http/PXHttpClient.java [hunk 1]
```diff
@@ -10,9 +10,11 @@
 import com.perimeterx.models.httpmodels.RiskResponse;
 import com.perimeterx.utils.Constants;
 import com.perimeterx.utils.JsonUtils;
+import com.sun.corba.se.impl.orbutil.threadpool.TimeoutException;
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/http/PXHttpClient.java]: Why did you add imports you don't use?

File: src/main/java/com/perimeterx/internals/PXCaptchaValidator.java [hunk 2]
```diff
@@ -33,16 +36,26 @@ public boolean verify(PXContext context) throws PXException {
...
                 return true;
             }
+            context.setBlockReason(BlockReason.SERVER);
             return false;
-        } catch (IOException e) {
-            return false;
+        } catch (Exception e) {
+            context.setRiskRtt(System.currentTimeMillis() - startRiskRtt);
+            //Handle timeout exception, else it a different error
+            if (e instanceof ConnectTimeoutException){
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/internals/PXCaptchaValidator.java]: You should `catch` this exception and do whatever you need, before catching other exceptions. It's better practice than checking its instance.

Also, don't throw an exception in a `catch` block.

File: src/main/java/com/perimeterx/internals/PXS2SValidator.java [hunk 3]
```diff
@@ -15,25 +20,44 @@
...
+            if (pxContext.getScore() <= pxConfiguration.getBlockingScore()){
+                pxContext.setPassReason(PassReason.S2S);
+                return true;
+            }
+            pxContext.setBlockReason(BlockReason.SERVER);
+            return false;
+        } catch (Exception e) {
+            pxContext.setRiskRtt(System.currentTimeMillis() - startRiskRtt);
+            //Handle timeout exception, else it a different error
+            if (e instanceof ConnectTimeoutException){
```
Comment (yaronpx) [inline src/main/java/com/perimeterx/internals/PXS2SValidator.java]: Again, catch the specific exception you want in a different block

**Comment by nitzanpx** [comment]
@pxaviad Can you please review
